dvc/tree/azure.py

```diff
@@ -29,21 +30,77 @@ class AzureTree(BaseTree):
     def __init__(self, repo, config):
         super().__init__(repo, config)
 
+        self._account_url = None
         url = config.get("url", "azure://")
         self.path_info = self.PATH_CLS(url)
 
         if not self.path_info.bucket:
             container = self._az_config.get("storage", "container_name", None)
             self.path_info = self.PATH_CLS(f"azure://{container}")
 
+        self._get_string_token_credential(config)
+
+        if not (self._conn_str or self._credential):
+            from azure.identity import (
+                AzureCliCredential,
+                ChainedTokenCredential,
+            )
+
+            if not self._storage_account:
+                logging.warning(
```

Comment:
if we must enter it, let's raise an exception here?

Reply:
I ran into an issue with throwing exceptions. This is what I've mentions to @efiop in a comment above. Any idea how to get over this?
"
Raising an exception broke a number of tests, one of which was dvc\tests\unit\dependency\test_azure.py because somewhere in the execution, a config object is created, one which I can't get a handle on, so the exception is raised and the test fails... this way is more in keeping with the previous behaviour where the user will just fail to authenticate against the remote and there will be logging if they wish to find more information about why
"